### ISSUE TITLE
Upload symbols to symbol server

### ISSUE THREAD

**Issue by sbomer** [dotnet/coreclr#21941]
With https://github.com/dotnet/coreclr/pull/21769, we publish symbol packages to a blob feed, but they don't yet get uploaded to a symbol server.

- The arcade migration docs mention that `IsShipping` packages have symbols published to the symbol server by default.
- The onboarding samples do pass access tokens for the symbol server in the publish command: `/p:DotNetSymbolServerTokenMsdl=$(microsoft-symbol-server-pat)` `/p:DotNetSymbolServerTokenSymWeb=$(symweb-symbol-server-pat)`.
- Yet the default arcade publish logic appears not to upload to symweb using the SymbolUploader package unless UsingToolSymbolUploader is true, which isn't the case in any of the samples.

Corefx has additional symbol publishing logic using the SymbolUploader package.

**Comment by sbomer** [comment]
From @tmat:

This needs a bit cleanup.
 
Eventually, the build shouldn’t be publishing symbols (or any other assets) directly. All it should do is publish to BAR. Then, once implemented, the release pipeline will publish the symbols that were uploaded to BAR.
 
Since the release pipeline does not work yet, you might want to publish symbols and other assets directly from the build if you need them to be published.
 
You have a couple of options:
1. Some repos (dotnet/symreader, dotnet/roslyn, etc.) have a separate YAML pipeline definition that provides the functionality that is currently missing from the release pipeline.
We use this definition for official builds:
https://github.com/dotnet/symreader/blob/master/azure-pipelines-microbuild.yml
In this definition we do not publish assets to BAR. We publish symbols directly to the symbol servers, packages to myget.org, VSIXes to VSTS drop, etc.
This is where the access tokens are specified:
https://github.com/dotnet/symreader/blob/master/azure-pipelines-microbuild.yml#L23-L24
The Arcade YAML build definition does instead publish to blob feed:
https://github.com/dotnet/symreader/blob/master/azure-pipelines.yml#L46-L47

2. If you don’t want 2 YAML definitions you can combine them together passing both symbol server access tokens and blob feed key and URL, but that’s a bit hacky. You’ll also need to specify `/p:PublishToSymbolServer=true` to force direct symbol publishing. This is kind of odd because now you’re gonna publish the symbols of each build to both BAR and symbol servers.

(thanks @tmat!)

**Comment by RussKeldorph** [comment]
@dagood Isn't core-setup publishing coreclr's symbols to the symbol server?  Are you aware of any work that has to be done in this area to ship 3.0?  I believe we could clean things up, but if we're functionally correct for 3.0, I'd rather defer this work since it will be significantly affected by later changes.

/cc @mikem8361

**Comment by dagood** [comment]
> Isn't core-setup publishing coreclr's symbols to the symbol server?

Yep. For anything Core-Setup redists, it also redists the symbols in the corresponding symbol package. Symbol packages are published for daily builds, and during releases they are reuploaded with a longer expiration date.

> Are you aware of any work that has to be done in this area to ship 3.0?

I believe per-repo symbol publish implementations need to be removed in favor of Arcade's publishing stages to make sure we're capable of performing servicing builds that don't publish to public locations (like the MSDL symbol server). I don't know the specific timeline for this change, though.

/cc @JohnTortugo
